src/main.d

```diff
@@ -74,18 +75,36 @@ void main(string[] args)
                 .map!(depsFile => readDependencies(File(depsFile)))
                 .join;
         actualDependencies = actualDependencies.sort.uniq.array;
+
         if (!targetFiles.empty)
         {
             import uml : read;
 
-            uint count = 0;
+            bool errored = false;
             Dependency[] targetDependencies = null;
 
             foreach (targetFile; targetFiles)
                 targetDependencies ~= read(File(targetFile).byLine);
 
             if (!transitive)
                 targetDependencies.transitiveClosure;
+
+            bool canDepend(const string client, const string supplier)
+            {
+                // a -> b allows a.x -> b.y, unless there's a dependency a.x -> [not a].* or [not b].* -> b.y
```

Comment:
Do we really want this?
If we want to be specific for `c` with `a.x -> c.z` and `c.z -> b.y`, then we can no longer be less specific for `a -> b`?

Reply:
The intuition is that "any arrows that pierce the boundary of `a` remove the opacity of `a`". It may be misleading to have an arrow pointing into `a` and have further modules in `a` allowed to do undeclared imports anyways.

Besides, you're the one who asked for this. :)

Reply:
My idea was: if there are specific dependencies like `a.x -> b.y`, then the general dependency `a -> b` (from another diagram) shall be removed from the target dependencies. The remaining target dependencies shall be used to match prefixes of the actual dependencies.

(So `a.x -> c.z` would not effect `a -> b`.)

My hope is: the corresponding implementation is much closer to "can still be understood next month" than the current one.

Reply:
I think the most important feature of the implementation is understanding what it's supposed to accomplish. I haven't understood that with yours yet.